recipes/python/python.inc

```diff
@@ -1,49 +1,266 @@
-# The machine needs to be patched
-# Original patch from:
-# http://www.trevorbowen.com/2013/10/07/cross-compiling-python-2-7-5-for-embedded-linux/
-# http://bugs.python.org/file31991/Python-2.7.5-xcompile.patch
-# Note the patch has been modified!
-SRC_URI:>machine = " file://Python-${PV}-xcompile.patch;striplevel=3"
+# When building for machine target, include the full
+# dependency list
+DEPENDS:>machine = " \
+    native:python${MAJOR_PV} \
+    native:python${MAJOR_PV}-dev \
+    native:python${MAJOR_PV}-distutils \
+    native:python${MAJOR_PV}-pgen \
+    native:python${MAJOR_PV}-freeze-importlib \
+    libffi \
+    libsqlite3 \
+    libreadline \
+    libm \
+    libz \
+    libexpat \
+    libpthread \
+    libssl \
+    liblzo2 \
+    libbz2 \
+    libncurses \
+    libcrypto \
+"
 
-# Python is in caps!
+SRC_URI = " \
+    https://www.python.org/ftp/python/${PV}/Python-${PV}.tgz \
+    file://libffi.patch \
+"
+
+SRC_URI:>machine = " \
+    file://host-tools.patch \
+    file://include-dirs.patch \
+"
 S = "${SRCDIR}/Python-${PV}"
 
-EXTRA_OEMAKE_COMPILE:native="python Parser/pgen"
-EXTRA_OECONF:machine = "\
-  --disable-ipv6\
-  ac_cv_file__dev_ptmx=no\
-  ac_cv_file__dev_ptc=no\
+# Ensure that we are using the system copy of various
+# libraries (expat, zlib, libffi, and libmpdec),
+# rather than copies shipped in the tarball
+do_unpack[postfuncs] += "do_unpack_rm_deps"
+do_unpack_rm_deps() {
+    rm -r ${S}/Modules/expat
+    rm -r ${S}/Modules/zlib
+    rm -r ${S}/Modules/_ctypes/{darwin,libffi}*
+
+    # libmpdec is only bundled with python3
+    rm -rf ${S}/Modules/_decimal/libmpdec
+}
+
+EXTRA_OECONF:>machine = " \
+    ac_cv_file__dev_ptmx=yes \
+    ac_cv_file__dev_ptc=no \
+    --with-threads \
+    --with-computed-gotos \
+    --with-dbmliborder=ndbm:bdb:gdbm \
+    --with-system-expat \
+    --with-system-ffi \
+    --with-system-libmpdec \
+    --without-ensurepip \
+    --without-cxx-main \
+    --without-doc-strings \
+    --enable-shared \
+    --enable-ipv6 \
 "
 
-# Make sure to include pgen and python from native into the package
-do_install[postfuncs] += "${DO_INSTALL_EXTRA_FILES}"
-DO_INSTALL_EXTRA_FILES = ""
-DO_INSTALL_EXTRA_FILES:native = "do_install_extra_files"
-do_install_extra_files () {
-	install -m 0755 ${S}/Parser/pgen \
-		${D}/bin/pgen
+EXTRA_OECONF:>native = " \
+    --enable-shared \
+    --with-system-expat \
+    --with-system-ffi \
+    --with-system-libmpdec \
+    --without-cxx-main \
+    --with-expat=none \
+    --disable-sqlite3 \
+    --disable-tk \
+    --disable-curses \
+    --disable-codecs-cjk \
+    --disable-nis \
+    --disable-unicodedata \
+    --disable-dbm \
+    --disable-gdbm \
+    --disable-bsddb \
+    --disable-test-modules \
+    --disable-bz2 \
+    --disable-zlib \
+    --disable-ssl \
+"
+
+# Wrap the autotools configure function to skip exports
+# when compiling for native, and to set exports when
+# cross compiling
+CONFIGURE_FUNC = "do_configure_autotools"
```

Comment:
Instead of the two wrapper functions, why not just amend LDFLAGS and CFLAGS (machine/native specific) and add PGEN_FOR_BUILD and FREEZE_IMPORTLIB_FOR_BUILD to EXTRA_OECONF?